cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -75,12 +75,29 @@ def update_nodes(strategy, nodelist)
     end
   end
 
+  def get_all_queues
+    # Get all queue names from the cluster config
+    config = YAML.safe_load(File.read(node['cluster']['cluster_config_path']))
```

Comment:
config read is already done in line 124, would be possible to do it once?

Reply:
Done, doing that by passing config to get_all_queues(config) and get_queues_with_changes(config)